### PR TITLE
Solr Petset

### PR DESCRIPTION
**Pull request by sekka1**
Adding in a Solr petset

### CODE REVIEW

File: incubator/solr/solr.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#
+# Docker usage: https://hub.docker.com/_/solr/
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:2]: not sure why this comment is here.

File: incubator/solr/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+Solr
+=============
+This example runs Solr through a petset.
+
+* This requires the zookeeper petset.
+* Uses the official Solr docker image
+* Creates persistent disks for the solr data
```
Comment (chrislovecnm) [inline incubator/solr/README.md:7]: Need more information for new users on what PV is.  Just a short sentence.

File: incubator/solr/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+Solr
+=============
+This example runs Solr through a petset.
+
+* This requires the zookeeper petset.
```
Comment (chrislovecnm) [inline incubator/solr/README.md:5]: @viglesiasce should we be providing a link to these??  Kinda torn.

File: incubator/solr/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+* Uses the official Solr docker image
```
Comment (chrislovecnm) [inline incubator/solr/README.md:6]: Please provide link here to the github repo with the image.

File: incubator/solr/solr.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  template:
+    metadata:
+      labels:
+        app: solr
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      terminationGracePeriodSeconds: 0
+      containers:
+      - name: solr
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:37]: We need http://kubernetes.io/docs/user-guide/compute-resources/ defined.  And will need Java Heap settings here as well.  What settings does the container allow for?  What settings does `/opt/solr/bin/solar` pick up?  How much memory should we launch with?

File: incubator/solr/solr.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        image: solr:6.2.0
+        # solr flags: https://cwiki.apache.org/confluence/display/solr/Solr+Start+Script+Reference
+        command:
+         - "/bin/sh"
+         - "-c"
+         - "/opt/solr/bin/solr start -f -z zoo-0.zk.default.svc.cluster.local:2181,zoo-1.zk.default.svc.cluster.local:2181,zoo-2.zk.default.svc.cluster.local:2181"
+        ports:
+        - containerPort: 8983
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:45]: Is 8983 the only port we need?  Been a LONG time since I used SOLR 😀

File: incubator/solr/solr.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    spec:
+      terminationGracePeriodSeconds: 0
+      containers:
+      - name: solr
+        image: solr:6.2.0
+        # solr flags: https://cwiki.apache.org/confluence/display/solr/Solr+Start+Script+Reference
+        command:
+         - "/bin/sh"
+         - "-c"
+         - "/opt/solr/bin/solr start -f -z zoo-0.zk.default.svc.cluster.local:2181,zoo-1.zk.default.svc.cluster.local:2181,zoo-2.zk.default.svc.cluster.local:2181"
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:43]: Can we use golang templating or env vars for the zoo keeper names?

File: incubator/solr/solr.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    name: solr
+  # solr-*.solr.default.svc.cluster.local
+  clusterIP: None
+  selector:
+    app: solr
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: solr
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:24]: Can we add template values for namespace options?

File: incubator/solr/solr.yaml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  labels:
+    app: solr
+spec:
+  ports:
+  - port: 8983
+    protocol: TCP
+    name: solr
+  # solr-*.solr.default.svc.cluster.local
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:16]: Can we cleanup the comment?

File: incubator/solr/solr.yaml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    metadata:
+      labels:
+        app: solr
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      terminationGracePeriodSeconds: 0
+      containers:
+      - name: solr
+        image: solr:6.2.0
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:38]: Can we templatize the name of the image?

File: incubator/solr/solr.yaml [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  volumeClaimTemplates:
+  - metadata:
+      name: datadir
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
```
Comment (chrislovecnm) [inline incubator/solr/solr.yaml:60]: Can we make the storage size a parameter?

**Comment by viglesiasce** [comment]
Hey @sekka1!

This isnt formatted properly as a chart. For more info on the chart structure look at:
https://github.com/kubernetes/helm/blob/master/docs/charts.md

You can get the scaffolding built for you by running:

```
helm create solr
```

**Comment by sekka1** [comment]
@viglesiasce will format it correctly.  Going to get to this next week.  Trying to find time.  Ill have to figure this out.  I have ActiveMQ and Mongo which I have running in a petset also.

**Comment by viglesiasce** [comment]
Awesome! We have another submission for Mongo over in #74.

File: incubator/solr/solr.yaml [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  name: solr
+spec:
+  serviceName: "solr"
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: solr
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
```
Comment (sroze) [inline incubator/solr/solr.yaml:33]: Should this really be in here?

File: incubator/solr/solr.yaml (hunk 1 above)
Comment (sekka1) [reply incubator/solr/solr.yaml:2]: Yeah, i think i should take these out.  It was really just my own notes on where i got the image from.  I think it is easy enough to just look at the image and know it was from docker hub.

File: incubator/solr/README.md (hunk 3 above)
Comment (sekka1) [reply incubator/solr/README.md:5]: I think once i change this into the helm format i can remove these comments and just put the zk as a dependency.

File: incubator/solr/solr.yaml (hunk 7 above)
Comment (sekka1) [reply incubator/solr/solr.yaml:43]: Yes, good idea.  This will make it much more dynamic.  Will change this.

**Comment by sekka1** [comment]
I signed it!

**Comment by viglesiasce** [comment]
Closing. Please open a new PR when this is made into a chart. Also might be good to make it a StatefulSet.

**Pull request closed without merging**